### PR TITLE
feat: Mobile Navigation

### PR DESCRIPTION
**Pull request by premiare**
Added a mobile navigation menu up until `md:` breakpoint `(768px)`

Contains the following:
- Basic slide-out from right mobile menu - (this can be changed to a top down animation)
- framer-motion animated links
- Reset menu behaviour if device width changes, (resize or tablet orientation changes)
- Clicking links close the menu

Preview:
https://www.loom.com/share/3c06b530b8724f8ba385ff2718f86544?sid=2f4d856d-323f-480c-8a18-0c96016d13e3

Build output:
```
Route (app)                                                     Size     First Load JS
┌ ○ /                                                           13.4 kB         181 kB
├ ● /[content]                                                  709 B          86.3 kB
├ ○ /blog                                                       137 B          78.4 kB
├ ● /blog/[post]                                                427 B          91.9 kB
├   ├ /blog/blog%2Fannouncing-open-source-docusign-alternative
├   ├ /blog/blog%2Fbuilding-documenso-pt1
├   └ /blog/blog%2Fmanifest
├ λ /claimed                                                    1.83 kB        85.9 kB
├ ○ /oss-friends                                                4.8 kB          136 kB
└ ○ /pricing                                                    8.12 kB         171 kB
+ First Load JS shared by all                                   78.3 kB
  ├ chunks/1dd3208c-5e49a948733d533c.js                         50.5 kB
  ├ chunks/575-b3f17f0dc55cfc6c.js                              25.7 kB
  ├ chunks/main-app-28d2072fdacf0241.js                         216 B
  └ chunks/webpack-4ce4dcdc46caa315.js                          1.79 kB

Route (pages)                                                   Size     First Load JS
┌ ○ /404                                                        178 B          75.8 kB
├ λ /api/claim-plan                                             0 B            75.6 kB
└ λ /api/stripe/webhook                                         0 B            75.6 kB
+ First Load JS shared by all                                   75.6 kB
  ├ chunks/framework-0857c1c24dbfb1d6.js                        45 kB
  ├ chunks/main-cdde65adb9216d0b.js                             28.6 kB
  ├ chunks/pages/_app-12a7bb3ca92362ee.js                       195 B
  └ chunks/webpack-4ce4dcdc46caa315.js                          1.79 kB

λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
```

Note: haven't tested on physical mobiles yet, just browser testing.

### CODE REVIEW

File: apps/marketing/src/components/(marketing)/mobile-navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    closed: { opacity: 0, y: 0, x: shouldReduceMotion ? 0 : 60, transition: { duration: 0.2 } },
+    exit: {
+      opacity: 0,
+      y: 0,
+      x: shouldReduceMotion ? 0 : 60,
+      transition: { duration: 0.2 },
+    },
+  };
+
+  // used for testing alternate animations
```
Comment (premiare) [inline apps/marketing/src/components/(marketing)/mobile-navigation.tsx:73]: If you wish to test alternative animations, vertical or horizontal, that's what these comments are for. Can be easily removed, no dramas.

**Comment by Mythie** [comment]
Thanks for your contribution Brayden! 🚀 

shawn-the-sheep

**Comment by Mythie** [comment]
This was just us appreciating the effort of the community. Bounties are created and labelled accordingly 😅

**Comment by rishi-raj-jain** [comment]
Ah aight, thanks for the quick response, @Mythie!

**Comment by Mythie** [comment]
roz-monsters

**Pull request merged**